# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/satyamgupta04/week1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane: Refresh / Content Opportunity Scoring

I chose the **Refresh / Content Opportunity Scoring** lane. My goal is to identify which pages should be reviewed first for actions such as **refresh, expansion, protection, monitoring, or pruning**.

### Why I Chose This Lane

I chose this lane because the starter notebooks showed that a **learned model could rank declining or at-risk pages more effectively than a simple hand-written rule**. This suggests that search, content, and engagement signals may help prioritize limited review time more effectively.

### Project Goal

This project will focus on **decision support, not automatic decisions**. The final goal is to produce a **ranked list of pages with clear reason codes**, so that a human reviewer can decide what action to take.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

Which content pages should be prioritized for **human review** based on observable search, content, and engagement signals?

### Unit of Analysis

The unit of analysis is a **content page**. Each page will be evaluated using available signals such as:

- Impressions
- Clicks
- CTR
- Average position
- Content age
- Freshness
- Engagement

### Decision

The system should help decide **which pages should be reviewed first** when the content team has limited time and cannot review every page.

### Output

The expected output is a **ranked list of pages**, where higher-ranked pages represent stronger candidates for review. The recommendations should also include **reason codes** explaining why a page was prioritized.

### Action

Based on the recommendation and human review, a content team could decide to:

- **Refresh** a page
- **Expand** its content
- **Protect** a well-performing page
- **Monitor** its performance
- Potentially **prune** a page

### Cost of a Wrong Recommendation

A **false positive** could waste time and resources by prioritizing a page that did not need attention.

A **false negative** could be more costly because the system might miss a page with an important opportunity or genuine decline.

### Why Data or ML Can Help

Reviewing a large number of pages manually is difficult. **Data and machine learning can combine multiple observable signals** to help prioritize pages more efficiently.

The starter results also suggest that a learned ranking may outperform a simple fixed rule. However, the model should be used to **support human decisions rather than automatically deciding that a page must be changed**.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [9]:
import pandas as pd

url = "https://raw.githubusercontent.com/satyamgupta04/week1/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Dataset loaded successfully!")
print("Total pages:", len(df))
df.head()

Dataset loaded successfully!
Total pages: 30000


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [10]:
# 1. Total pages
total_pages = len(df)

# 2. Pages with a declining trend
declining_pages = (df["trend_direction"] == "down").sum()
declining_pct = (declining_pages / total_pages) * 100

# 3. Declining pages that still have search demand
declining_with_demand = (
    (df["trend_direction"] == "down") &
    (df["impressions_90d"] >= 100)
).sum()

print("Total pages:", total_pages)
print("Declining pages:", declining_pages)
print(f"Percentage declining: {declining_pct:.2f}%")
print("Declining pages with 100+ impressions:", declining_with_demand)

Total pages: 30000
Declining pages: 16262
Percentage declining: 54.21%
Declining pages with 100+ impressions: 13152


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*


### What I can claim

From the starter dataset, I observed that 16,262 out of 30,000 pages (54.21%) have a declining trend, and 13,152 declining pages still have at least 100 impressions. This suggests that there is a meaningful set of pages that could be prioritized for human review using observable search, content, and engagement signals.

My work can support decisions about which pages should be reviewed first.

### What I cannot claim

I cannot claim that refreshing a declining page will cause its traffic or search performance to improve. Declines may also be influenced by seasonality, consolidation, or normal fluctuations.

The system should be treated as a decision-support tool for prioritizing human review, not as proof of what causes Google rankings or as an automatic decision about which pages must be changed.




## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.